runtime/src/main/jni/ObjectManager.cpp

```diff
@@ -513,6 +517,18 @@ void ObjectManager::MarkReachableObjects(Isolate* isolate, const Local<Object>&
         } // for
 
     } // while
+
+    if (frame.check()) {
+        auto cls = fromJsInfo->ObjectClazz;
+        JEnv env;
+        jclass clsClazz = env.GetObjectClass(cls);
+        jmethodID methodId = env.GetMethodID(clsClazz, "getSimpleName", "()Ljava/lang/String;");
```

Comment:
consider caching the jclass and jmethodID in static variables in the Frame class if the frame.check() passes (profiling is enabled)